Rare High Traffic Deadlock situation in capture test

In CI run https://github.com/els0r/goProbe/actions/runs/6459269585/job/17534700503#step:7:148 the high traffic / throughput test guarding against potential deadlocks triggered once (without any obvious change that might have caused it). A re-run failed to reproduce the issue, so it's probably a sporadic race condition or similar.

```
--- PASS: TestLowTrafficDeadlock (1.53s)
    --- PASS: TestLowTrafficDeadlock/0_packets (0.30s)
    --- PASS: TestLowTrafficDeadlock/1_packets (0.30s)
    --- PASS: TestLowTrafficDeadlock/10_packets (0.31s)
    --- PASS: TestLowTrafficDeadlock/100_packets (0.31s)
    --- PASS: TestLowTrafficDeadlock/1000_packets (0.31s)
=== RUN   TestHighTrafficDeadlock
    capture_test.go:370: potential deadlock situation on rotation logic
--- FAIL: TestHighTrafficDeadlock (10.36s)
```

--- a/pkg/capture/capture_test.go
+++ b/pkg/capture/capture_test.go
@@ -307,9 +307,8 @@ func testDeadlockLowTraffic(t *testing.T, maxPkts int) {
 		select {
 		case err := <-errChan:
 			doneChan <- err
-			break
 		case <-time.After(10 * time.Second):
-			doneChan <- errors.New("potential deadlock situation on rotation logic")
+			doneChan <- errors.New("potential deadlock situation on rotation logic (no termination confirmation received from mock source)")
 		}
 
 		require.Nil(t, mockC.close())
@@ -317,8 +316,8 @@ func testDeadlockLowTraffic(t *testing.T, maxPkts int) {
 
 	require.Nil(t, <-doneChan)
 
-	if time.Since(start) > 2*time.Second {
-		t.Fatalf("potential deadlock situation on rotation logic")
+	if time.Since(start) > 5*time.Second {
+		t.Fatalf("potential deadlock situation on rotation logic (test took %v)", time.Since(start))
 	}
 }
 
@@ -345,6 +344,8 @@ func testDeadlockHighTraffic(t *testing.T) {
 	start := time.Now()
 	doneChan := make(chan error)
 	time.AfterFunc(100*time.Millisecond, func() {
+
+		t.Logf("starting roation loops after %v", time.Since(start))
 		for i := 0; i < 20; i++ {
 			mockC.lock()
 			mockC.rotate(ctx)
@@ -352,22 +353,24 @@ func testDeadlockHighTraffic(t *testing.T) {
 			time.Sleep(10 * time.Millisecond)
 		}
 		mockSrc.Done()
+		t.Logf("roation loops done after %v", time.Since(start))
 
 		select {
 		case err := <-errChan:
 			doneChan <- err
-			break
 		case <-time.After(10 * time.Second):
-			doneChan <- errors.New("potential deadlock situation on rotation logic")
+			doneChan <- errors.New("potential deadlock situation on rotation logic (no termination confirmation received from mock source)")
 		}
 
 		require.Nil(t, mockC.close())
 	})
 
-	require.Nil(t, <-doneChan)
+	doneElem := <-doneChan
+	t.Logf("received signal from rotation Goroutine after %v: %v", time.Since(start), doneElem)
+	require.Nil(t, doneElem)
 
-	if time.Since(start) > 2*time.Second {
-		t.Fatalf("potential deadlock situation on rotation logic")
+	if time.Since(start) > 5*time.Second {
+		t.Fatalf("potential deadlock situation on rotation logic (test took %v)", time.Since(start))
 	}
 }
 

--- a/pkg/capture/capture_test.go
+++ b/pkg/capture/capture_test.go
@@ -297,12 +297,15 @@ func testDeadlockLowTraffic(t *testing.T, maxPkts int) {
 	start := time.Now()
 	doneChan := make(chan error)
 	time.AfterFunc(100*time.Millisecond, func() {
+
+		t.Logf("starting roation loops after %v", time.Since(start))
 		for i := 0; i < 20; i++ {
 			mockC.lock()
 			mockC.rotate(ctx)
 			mockC.unlock()
 			time.Sleep(10 * time.Millisecond)
 		}
+		t.Logf("roation loops done after %v", time.Since(start))
 
 		select {
 		case err := <-errChan:
@@ -316,7 +319,7 @@ func testDeadlockLowTraffic(t *testing.T, maxPkts int) {
 
 	require.Nil(t, <-doneChan)
 
-	if time.Since(start) > 5*time.Second {
+	if time.Since(start) > 10*time.Second {
 		t.Fatalf("potential deadlock situation on rotation logic (test took %v)", time.Since(start))
 	}
 }
@@ -358,18 +361,16 @@ func testDeadlockHighTraffic(t *testing.T) {
 		select {
 		case err := <-errChan:
 			doneChan <- err
-		case <-time.After(10 * time.Second):
+		case <-time.After(30 * time.Second):
 			doneChan <- errors.New("potential deadlock situation on rotation logic (no termination confirmation received from mock source)")
 		}
 
 		require.Nil(t, mockC.close())
 	})
 
-	doneElem := <-doneChan
-	t.Logf("received signal from rotation Goroutine after %v: %v", time.Since(start), doneElem)
-	require.Nil(t, doneElem)
+	require.Nil(t, <-doneChan)
 
-	if time.Since(start) > 5*time.Second {
+	if time.Since(start) > 30*time.Second {
 		t.Fatalf("potential deadlock situation on rotation logic (test took %v)", time.Since(start))
 	}
 }

